packages/syft/src/syft/service/project/project.py

```diff
@@ -107,14 +128,49 @@ def valid(self) -> Union[SyftSuccess, SyftError]:
         except Exception as e:
             return SyftError(message=f"Failed to validate message. {e}")
 
+    def valid_descendant(
+        self, project: UID, prev_event: Optional[Self]
+    ) -> Union[SyftSuccess, SyftError]:
+        valid = self.valid
+        if not valid:
+            return valid
+
+        if prev_event:
+            prev_event_id = prev_event.id
+            prev_event_hash = hash(prev_event)
```

Comment:
MAYBE: We should use the `.event_hash` attribute, as I think this would also has the signature of the previous event